### ISSUE TITLE
Blank highlight when search matches against an anchor

### ISSUE THREAD

**Issue by aandis** [typesense/typesense-docsearch.js#8]
## Description

Blank highlight when search matches against an anchor

## Steps to reproduce

Config
```
{
  "start_urls": [
    {
      "url": "https://example.com/"
    }
  ],
  "selectors": {
    "default": {
      "lvl0": {
        "selector": "h4"
      },
      "text": "article p, article td:last-child"
    }
  }
}
```

Markup
```
<div>
    <h4 class="anchor anchorWithStickyNavbar_LWe7" id="automatic-date-field-template">Automatic Date Field Template<a class="hash-link" href="#automatic-date-field-template" title="Direct link to heading">​</a></h4>
    <p>Automatic Date Field Template are a type of field template ...</p>
</div>
```

Search term - `Automatic Date Field Template`

Result - 
![Screenshot from 2023-03-05 23-18-03](https://user-images.githubusercontent.com/4983347/222977066-c390d137-376a-49e4-a505-09d2d230d644.png)

Erroneous result json
```
{
  "group_key": [
    "https://example.com/glossary/#automatic-date-field-template"
  ],
  "hits": [
    {
      "document": {
        "anchor": "automatic-date-field-template",
        "hierarchy": {
          "lvl0": "Automatic Date Field Template ​",
          "lvl1": null,
          "lvl2": null,
          "lvl3": null,
          "lvl4": null,
          "lvl5": null,
          "lvl6": null
        },
        "hierarchy.lvl0": "Automatic Date Field Template ​",
        "id": "13",
        "type": "lvl0",
        "url": "https://example.com/glossary/#automatic-date-field-template"
      },
      "highlight": {
        "hierarchy": {
          "lvl0": "Automatic Date Field Template ​",
          "lvl1": null,
          "lvl2": null,
          "lvl3": null,
          "lvl4": null,
          "lvl5": null,
          "lvl6": null
        }
      },
      "highlights": [],
      "text_match": 2314894167592927000,
      "text_match_info": {
        "best_field_score": "4419510927360",
        "best_field_weight": 15,
        "fields_matched": 1,
        "score": "2314894167592927353",
        "tokens_matched": 4
      }
    }
  ]
}
```

Retickted from https://typesense-community.slack.com/archives/C01P749MET0/p1677869631345769?thread_ts=1677869266.139009&cid=C01P749MET0

**Comment by jasonbosco** [comment]
@aandis In this issue's title you mentioned that the highlights are blank when the search matches against an **`anchor`**, but in the PR, you've added highlights for **`lvl0`**. Could you clarify this? 

Also, could you share the full JSON response returned by Typesense instead of a partial one?

**Comment by aandis** [comment]
@jasonbosco by `anchor`, I meant the `h4` headings (anchor links) which are stored in `lvl0` as per my configuration. The issue here is that when the result contains a match only against `lvl0` and rest of the fields are `null`, it shows up as an empty result like you see in the screenshot.

**Comment by aandis** [comment]
Closed via #9
